### PR TITLE
docs: explain specific nixos configurations

### PR DESCRIPTION
**Pull request by yajo**
**Description of your changes:**
Explain the fix for https://github.com/rook/rook/issues/10932 to let NixOS users set up Rook more easily.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/pull/10942#discussion_r971577389

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR. I can't!
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

@moduon MT-1248

### CODE REVIEW

**Comment by parth-gr** [comment]
PLease add a signed-off in the commit message

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 1]
```diff
@@ -101,3 +101,50 @@ or choose a different Linux distribution.
...
+### NixOS
+
+When you use NixOS, the kernel modules will be found in the non-standard path `/run/current-system/kernel-modules/lib/modules/`,
+and they'll be symlinks to stuff located inside the also non-standard path `/nix`.
+
+For Rook Ceph containers to be able to load the required modules, they need read access to those locations.
+
+If you install Rook with Helm, add this to your `values.yaml`:
+
+```yaml
```
Comment (parth-gr) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:118]: @travisn how about also leaving this setting commented in the manifest files?

**Comment by yajo** [comment]
Signed.

**Comment by parth-gr** [comment]
DCO still fails, can you please fix that,

It shows the error
`Commit sha: [d074f75](https://github.com/rook/rook/pull/11029/commits/d074f754c9fefd04c14fab6a4c50ea36fc29c442), Author: Jairo Llopis, Committer: Jairo Llopis; Expected "Jairo Llopis [973709+yajo@users.noreply.github.com](mailto:973709+yajo@users.noreply.github.com)", but got "Jairo Llopis [jairo@moduon.team](mailto:jairo@moduon.team)".`

**Comment by parth-gr** [comment]
> DCO still fails, can you please fix that,
> 
> It shows the error `Commit sha: [d074f75](https://github.com/rook/rook/pull/11029/commits/d074f754c9fefd04c14fab6a4c50ea36fc29c442), Author: Jairo Llopis, Committer: Jairo Llopis; Expected "Jairo Llopis [973709+yajo@users.noreply.github.com](mailto:973709+yajo@users.noreply.github.com)", but got "Jairo Llopis [jairo@moduon.team](mailto:jairo@moduon.team)".`

Seems like not a correct sign-off

**Comment by yajo** [comment]
Let's see now.

**Review by parth-gr — APPROVED** [review]
Looks good to me.
Let @travisn review it once.

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 2]
```diff
@@ -101,3 +101,50 @@ or choose a different Linux distribution.
...
 enforced on newer kernels.
+
+## Distro Notes
+
+Specific configurations for some distributions.
+
+### NixOS
+
+When you use NixOS, the kernel modules will be found in the non-standard path `/run/current-system/kernel-modules/lib/modules/`,
+and they'll be symlinks to stuff located inside the also non-standard path `/nix`.
```
Comment (travisn) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:112]:
```suggestion
and they'll be symlinked inside the non-standard path `/nix`.
```

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 3]
```diff
@@ -101,3 +101,50 @@ or choose a different Linux distribution.
...
+
+For Rook Ceph containers to be able to load the required modules, they need read access to those locations.
+
+If you install Rook with Helm, add this to your `values.yaml`:
```
Comment (travisn) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:116]: How about if we leave the complex example settings out of this doc and just mention the setting names?
```suggestion
If you install Rook with Helm, uncomment the examples settings in `values.yaml`:
- `csiCephFSPluginVolume`
- `csiCephFSPluginVolumeMount`
- `csiRBDPluginVolume`
- `csiRBDPluginVolumeMount`
```

**Review by travisn — CHANGES_REQUESTED** [review]
For the commit message, let's just have a "docs:" prefix, something like this:
```
docs: explain specific nixos configurations
```

**Comment by yajo** [comment]
All attended.

**Comment by travisn** [comment]
> For the commit message, let's just have a "docs:" prefix, something like this:
> 
> ```
> docs: explain specific nixos configurations
> ```

Just one more nit on the commit message, thanks!

**Comment by yajo** [comment]
You got it!

**Review by travisn — APPROVED** [review]

**Pull request merged**